source/val/ValidationState.h

```diff
@@ -42,17 +43,24 @@
 
 namespace libspirv {
 
+class Function;
+class BasicBlock;
+
+static const uint32_t kInvalidId = 0x400000;
+
 // Info about a result ID.
-typedef struct spv_id_info_t {
+struct spv_id_info_t {
   /// Id value.
   uint32_t id;
   /// Type id, or 0 if no type.
   uint32_t type_id;
   /// Opcode of the instruction defining the id.
   SpvOp opcode;
+  /// The block where the id was defined. kInvalidId if module scoped
```

Comment:
Yeah, but my recent PR for psuedo-entry node cheats and uses Id 0 for the psuedo-entry node.   So please keep this one.  :-)